Text/AbbyyOCR/AbbyyOCR.cs

```diff
@@ -0,0 +1,257 @@
+			}
+
+			// Process image 
+			// You could also call ProcessDocumentAsync or any other processing method declared below
+			var resultUrls = await ProcessImageAsync(ocrClient, localFile);
+
+			//Get results - the first doc is a docx, second is a text file
+			using (var client = new WebClient())
+			{
+				waRecord.Data.Add("content", client.DownloadString(resultUrls[1].ToString()));
+			}
+
+			File.Delete(Path.GetTempPath() + "\\" + "temp_" + idx.ToString() + fileType);
+
+			waRecord.RecordId = record.RecordId;
+
+			bag.Add(waRecord);
+		}
+
+		private static IOcrClient GetOcrClient()
+		{
+			return new OcrClient(AuthInfo);
+		}
+
+		private static IOcrClient GetOcrClientWithRetryPolicy()
+		{
+			// Create service collection and configure our services
+			var services = ConfigureServices();
+			// Generate a provider
+			_serviceProvider = services.BuildServiceProvider();
+
+			var httpClientFactory = _serviceProvider.GetService<IHttpClientFactory>();
+			_httpClient = httpClientFactory.CreateClient(_httpClientName);
+
+			return new OcrClient(_httpClient);
+		}
+
+		private static ServiceCollection ConfigureServices()
+		{
+			var services = new ServiceCollection();
+
+			//Configure HttpClientFactory with retry handler
+			services.AddHttpClient(_httpClientName, conf =>
+			{
+				conf.BaseAddress = new Uri(AuthInfo.Host);
+				//increase the default value of timeout for the duration of retries
+				conf.Timeout = conf.Timeout + TimeSpan.FromSeconds(_retryCount * _delayBetweenRetriesInSeconds);
+			})
+				.ConfigurePrimaryHttpMessageHandler(() => new HttpClientHandler
+				{
+					PreAuthenticate = true,
+					Credentials = new NetworkCredential(AuthInfo.ApplicationId, AuthInfo.Password)
+				})
+				//Add  custom HttpClientRetryPolicyHandler with polly
+				.AddHttpMessageHandler(() => new HttpClientRetryPolicyHandler(GetRetryPolicy()));
+
+			//or you can use Microsoft.Extensions.DependencyInjection Polly extension
+			//.AddPolicyHandler(GetRetryPolicy());
+			return services;
+		}
+
+		private static IAsyncPolicy<HttpResponseMessage> GetRetryPolicy()
+		{
+			return HttpPolicyExtensions.HandleTransientHttpError()
+				//Condition - what kind of request errors should we repeat
+				.OrResult(r => r.StatusCode == HttpStatusCode.GatewayTimeout)
+				.WaitAndRetryAsync(
+					_retryCount,
+					sleepDurationProvider => TimeSpan.FromSeconds(_delayBetweenRetriesInSeconds),
+					(exception, calculatedWaitDuration, retries, context) =>
+					{
+						Console.WriteLine($"Retry {retries} for policy with key {context.PolicyKey}");
+					}
+				)
+				.WithPolicyKey("WaitAndRetryAsync_For_GatewayTimeout_504__StatusCode");
+		}
+
+		private static async Task<List<string>> ProcessImageAsync(IOcrClient ocrClient, string localFile)
+		{
+			var imageParams = new ImageProcessingParams
+			{
+				ExportFormats = new[] { ExportFormat.Docx, ExportFormat.Txt, },
+				Language = "English,Arabic,Hebrew",
```

Comment:
Should this be configurable?

Reply:
I mention this cofigurability in the Readme, so I think this is ok.